pkg/tenant/db/topics_test.go

```diff
@@ -72,8 +73,12 @@ func TestTopicValidate(t *testing.T) {
 	topic.Name = "otters;"
 	require.ErrorIs(t, topic.Validate(), db.ErrInvalidTopicName, "expected invalid name to be an error")
 
+	// Test name that is too long
+	topic.Name = strings.Repeat("a", 513)
+	require.ErrorIs(t, topic.Validate(), db.ErrTopicNameTooLong, "expected too long name to be an error")
+
 	// Valid topic
-	topic.Name = "otters"
+	topic.Name = "otters_are-cool"
```

Comment:
Since the validation is a bit complicated, I think we should use a test case pattern similar to the following.

```golang
testCases := struct{}{
    name string,
    err error
}{
    ("otters;", db.ErrInvalidTopicName},
    ("otters_are-cool", nil),
}

for i, tc := testCases {
    if tc.err == nil {
        require.NoError(topic.Validate(), fmt.Sprintf("expected no error for test case %d", i)
    } else {
        require.ErrorIs(tc.err, topic.Validate(), fmt.Sprintf("expected error for test case %d", i)
    }
}
```

That will allows us to keep track of all the edge cases, so can you add a story for a future sprint to revise the test and add some more edge cases?